src/main/thrift/parquet.thrift

```diff
@@ -833,6 +843,13 @@ struct ColumnMetaData {
    * filter pushdown.
    */
   16: optional SizeStatistics size_statistics;
+
+  /**
+   * The index into FileMetadata.schema (list<SchemaElement>) for this column.
+   * This implies that ColumnMetaData can be sparse in a rowgroup, if for example
+   * a column does not have any data pages in a rowgroup.
+   */
+  17: optional i32 schema_index;
```

Comment:
FWIW I accidentally discovered https://issues.apache.org/jira/browse/PARQUET-183 which can be fixed with this field.